libsupport/include/katana/Iterators.h

```diff
@@ -0,0 +1,187 @@
+#ifndef KATANA_LIBSUPPORT_KATANA_ITERATORS_H_
+#define KATANA_LIBSUPPORT_KATANA_ITERATORS_H_
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+#include <boost/iterator/iterator_facade.hpp>
+
+namespace katana {
+
+//TODO(amber): Move other iterators from libgalois to libsupport
+
+/**
+ * A Zip Iterator allows iterating over several containers simultaneously.
+ * A canonical example is sorting two containers together. A Zip iterator maintains
+ * a tuple of participating iterators and advances all of them together.
+ * Dereferencing yields a tuple of references for l-values and a tuple of values
+ * for r-values. See  code below:
+ * for example usage:
+
+ * NOTE: It is best to define comparator functions as template functions (or
+ * lambdas that take parameters as 'auto'), e.g., for std::sort(),
+ * because either left or right parameter to the comparator can be a
+ * tuple of references or tuple of values depending on the calling context.
+
+int main() {
```

Comment:
I don't know if Doxygen will like this formatting.

I suggest just using `///` on every line.

Reply:
done